src/views/Bridge/hooks/useSelectRoute.ts

```diff
@@ -196,34 +222,73 @@ export function useSelectRoute() {
         ),
         fromChain: selectedRoute.fromChain,
         toChain: toChainId,
+        externalProjectId,
       };
-      const route =
-        findNextBestRoute(
-          [
-            "fromChain",
-            "toChain",
-            isSwap ? "swapTokenSymbol" : "inputTokenSymbol",
-          ],
-          filterBy
-        ) ||
-        findNextBestRoute(["fromChain", "toChain"], filterBy) ||
-        findNextBestRoute(["fromChain", "toChain"], {
-          ...filterBy,
-          outputTokenSymbol: undefined,
-        }) ||
-        findNextBestRoute(["fromChain"], {
-          toChain: toChainId,
-        }) ||
-        findNextBestRoute(
-          ["toChain", isSwap ? "swapTokenSymbol" : "inputTokenSymbol"],
-          filterBy
-        ) ||
-        initialRoute;
+
+      // Try to find route with exact match first
+      let route = externalProjectId
+        ? findNextBestRoute(["toChain", "externalProjectId"], {
+            toChain: toChainId,
+            externalProjectId,
+          })
+        : findNextBestRoute(["toChain"], {
+            toChain: toChainId,
+            externalProjectId: undefined,
+          });
```

Comment:
I _think_ this is breaking some UX behavior. This should be 
```suggestion
        : findNextBestRoute(
          [
            "fromChain",
            "toChain",
            isSwap ? "swapTokenSymbol" : "inputTokenSymbol",
          ],
          filterBy
        );
```
I believe. 

The bug: I you select the from chain to anything other than Ethereum Mainnet and then change the destination chain, the origin chain changes back to Ethereum Mainnet

Reply:
https://github.com/across-protocol/frontend/pull/1360/commits/bb80c0b573b9fd3c15ed9468d04223844fe1a430